src/Plugins/Loader/AssemblyLoadContextBuilder.cs

```diff
@@ -148,21 +150,46 @@ public AssemblyLoadContextBuilder PreferLoadContextAssembly(AssemblyName assembl
         /// <returns>The builder.</returns>
         public AssemblyLoadContextBuilder PreferDefaultLoadContextAssembly(AssemblyName assemblyName)
         {
-            if (assemblyName.Name == null || _defaultAssemblies.Contains(assemblyName.Name))
+            // Lazy loaded references have dependencies resolved as they are loaded inside the actual Load Context.
+            if (_lazyLoadReferences)
             {
-                // base cases
+                if (assemblyName.Name != null && !_defaultAssemblies.Contains(assemblyName.Name))
+                {
+                    _defaultAssemblies.Add(assemblyName.Name);
+                    var assembly = _defaultLoadContext.LoadFromAssemblyName(assemblyName);
```

Comment:
This seems like a non-lazy action to me. What do you think about using the "IsLazyLoaded" option to completely defer *all* loading?

Reply:
It's been a bit over 2 months and my memory of the exact technical details are a bit hazy but allow me to try and explain (might be a bit hard though).

The `lazy` aspect of `lazy loading` is actually the determination of nested transitive assemblies.

Deferring *all* loading is simply impossible; as the `ManagedLoadContext` needs to be aware of at least 1 level/depth of transitive types for any shared type.

Suppose you don't load the shared type into the default ALC inside `AssemblyLoadContextBuilder`; the transitive reference name `Test.Transitive` is never added to `_defaultAssemblies`.  The problem arises when you later try to use a Class/Method/Property or anything that requires JIT-ting which references `Test.Transitive`. `ManagedLoadContext.Load` is called and the load context isn't aware of the existence of the transitive dependency; it thus defers to loading from file path and unification fails.

Reply:
Makes senes, thanks for explaining.